Adds Sample Event for JSON and MD

rsahlstrom opened the pull request:
This is an example card from Tales of Arabian Nights coded
in json. I only did two of the six reactions.

The markdown format is not final but was rather just an
attempt to capture the contents of the card from the game.

rsahlstrom committed:
Adds Sample Event for JSON and MD

This is an example card from Tales of Arabian Nights coded
in json. I only did two of the six reactions.

The markdown format is not final but was rather just an
attempt to capture the contents of the card from the game.

rsahlstrom commented on events/json/lovesickEfreet.json:
```diff
@@ -0,0 +1,197 @@
+            {
+                "text": "At your words the Lovesick 'Efreet says, \"I have no patience with such foolish offers! Get out of my sight!\" You do so.",
+                "skills": {
+                    "none": {
```
Note the lack of a text property in this object. This story basically has no branching outcomes based on skills and just immediately gives out results. Instead of coming up with a unique structure for this, I just associated the reward with the none skills result and left out the text property. If we didn't want to make text optional, we could also do this structure as:

``` json
"skills": {
    "none": {
        "text": "",
        "rewards": [...]
    }
}
```

hhatfield replied:
I think it is ok to have an empty text field, but should not pass validation if it's missing one.

rsahlstrom replied:
I added an empty text field.

rsahlstrom commented:
Not sure I did the options and skills section of the stories right. I'm thinking they should maybe be arrays and we move their keys to a title property of the anonymous object. For example:

``` json
{
    "title": "Lovesick 'Efreet",
    "options": [
        {
            "title": "grovel",
            "text": "The Lovesick 'Efreet glares in response to your words. \"You seek only to gain profit at my expense, do you not?\"",
            "skills": [
                {
                    "title": "none",
                    "text": "The Lovesick 'Efreet will have no dealings with you; your conversation is seen by many, and your shame is great.",
                    "rewards": [
                        {
                            "type": "status",
                            "gain": "scorned"
                        }
                    ]
                },
                {
                    "title": "beguiling",
                    "mandatory": true,
                    "text": "Your confidence in your wit and tongue lead you to attempt to trick the Lovesick 'Efreet, when it is obvious that he is skeptical of your character. You persist long after sense would have dictated you withdraw, and he publicly humiliates you for your false dealing.",
                    "rewards": [
                        {
                            "type": "destiny",
                            "step": 1
                        },
                        {
                            "type": "skill",
                            "lose": "beguiling"
                        },
                        {
                            "type": "status",
                            "gain": "scorned"
                        }
                    ]
                }
            ]
        }
    ]
}
```

Thoughts?

rsahlstrom commented:
This still might need to be reworked. Some paragraphs allow for multiple skills or treasure cards to trigger them. Probably need to redo the skills node a bit to make it work.

Any feedback is still appreciated. This is just a rough draft so bash away at it.

rsahlstrom committed:
Changes Skills to Outcomes, adds prerequisites

Changes the skills object to the outcomes array. Array makes
more sense as there are any number of possible outcomes to a
story and outcomes are not always based on just skills.

The new prerequisites array allows you to list conditions that
must be meet for that outcome to happen. Currently, I see this
as you need to meet any one of the prerequisites but we will
probably need to come up with a method to allow for needing to
meet multiple prerequisites.

rsahlstrom commented on events/json/lovesickEfreet.json:
```diff
@@ -0,0 +1,240 @@
+                    {
+                        "prerequisites": [
+                            {
+                                "type": "skill",
```
If we needed to check for multiple prerequisites you could do something crazy like this:

``` json
"prerequisites": [
    {
        "type": "prerequisites",
        "value": [
            {
                "type": "skill",
                "value": "magic"
            },
            {
                "type": "treasure",
                "value": "flying carpet"
            }
        ]
    }
]
```

hhatfield commented on events/json/lovesickEfreet.json:
```diff
@@ -0,0 +1,240 @@
+{
+    "title": "Lovesick 'Efreet",
```
Probably need to add a "type" field so we can distinguish easily between the different card types

rsahlstrom replied:
Good call.

hhatfield commented on events/json/lovesickEfreet.json:
```diff
@@ -0,0 +1,240 @@
+                            {
+                                "type": "wealth",
+                                "step": 2,
+                                "max": "rich"
```
where would something like "rich" be defined?

rsahlstrom replied:
We'll probably need some kind of config file where you would define things like the wealth options, what score tracks you're using (here we use destiny and story), etc.
